src/cpp/include/openvino/genai/generation_config.hpp

```diff
@@ -132,7 +134,8 @@ class OPENVINO_GENAI_EXPORTS GenerationConfig {
     bool is_greedy_decoding() const;
     bool is_beam_search() const;
     bool is_multinomial() const;
-    bool is_speculative_decoding() const;
```

Comment:
why was it renamed? 

It's not source BW compatible

Reply:
Because, the name is not compatible with the new changes. So, we are providing answer for `is_assisting_generation` now, not for `is_specualtive_decoding`

Reply:
then we need to deprecate this function first, example:
https://github.com/openvinotoolkit/openvino.genai/blob/c77f7c93f5f82cde4988a8ef1b3ca204d3d6873d/src/cpp/include/openvino/genai/llm_pipeline.hpp#L124-L126